app/javascript/components/student/Editor.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { useState } from 'react'
+import { CodeEditor } from './editor/CodeEditor'
+import { TestRunSummary } from './editor/TestRunSummary'
+import fetch from 'isomorphic-fetch'
+
+export type Submission = {
+  testsStatus: TestRunStatus
+  uuid: string
+}
+
+export enum TestRunStatus {
+  PASS = 'pass',
+  FAIL = 'fail',
+  ERROR = 'error',
+  PENDING = 'pending',
+  OPS_ERROR = 'ops_error',
+}
+
+export function Editor({ endpoint }: { endpoint: string }) {
+  const [submission, setSubmission] = useState<Submission>()
```

Comment:
In strict typescript, this may fail, unless they changed the typing (realistically possible).

The type of submission isn't `Submission` but `Submission | undefined`.

Check if `strict` is true, and if yes, and this doesn't complain, ignore me. They changed the typing to automatically add `undefined` unless an initial value is given.

Reply:
It's set to strict in `tsconfig.json`.